### DISCUSSION TITLE
Failing when deploying into a testnet

### DISCUSSION THREAD

**Discussion by fricpto** [smartcontractkit/full-blockchain-solidity-course-py#1772] [Q&A] [answered]
when I run brownie run scripts/deploy.py --network rinkeby I get this result:
<img width="554" alt="Screenshot 2022-11-09 165854" src="https://user-images.githubusercontent.com/100000316/200943142-780765b8-62ec-411f-92d6-96ba0c66f0dc.png">
and this is my deploy.py code:
```Python
 from brownie import accounts, config, SimpleStorage, network


def deploy_simple_storage():
    account = get_account()
    simple_storage = SimpleStorage.deploy({"from": account})
    stored_value = simple_storage.retrieve()
    print(stored_value)
    transaction = simple_storage.store(15, {"from": account})
    transaction.wait(1)
    updated_stored_value = simple_storage.retrieve()
    print(updated_stored_value)


def get_account():
    if network.show_active() == "development":
        return accounts[0]
    else:
        return accounts.add(config["wallets"]["from_key"])


def main():
    deploy_simple_storage()
```

and this is my .env file code:
**Removed the Image by Maintainer to avoid security risk**

-python and my .env aren't giving me any errors or problems please can someone try to help to solve the issue here plus if I run on a Goerli testnet I'll get the same result.

**Comment by cromewar** [comment]
Hello @fricpto 
Rinkeby was deprecated so it won't work anymore, please give a check to the repo and see the minor changes made to work with Goerli.
Now, as you said is not working with  Goerli Either, please be sure to:

1. have added `dotenv: .env` to your brownie config.
2. have configured correctly your Infura endpoint.
3. Sometime you might like to also add the private key to your windows path variables.

Important Note, you have shared a screenshot with your private keys, please don't do so. You  should never share those with anyone.

**Comment by fricpto** [comment]
@cromewar what is the repo that I should check to see how can I work with Goerli eth and how to check the Infura endpoint (I have just copied and paste my project ID)
P.S: those private keys that I have shared I don't care about them because I never put money into the wallet that is connect to this private key.

**Reply by cromewar** [reply]
If I'm not wrong you are on lesson 5, so here you have the repo code: https://github.com/smartcontractkit/full-blockchain-solidity-course-py#lesson-5-brownie-simple-storage
And as I said be sure to have the private keys on your env variables, both on your `brownie-config.yaml` and you path (on windows you can add it trough advanced configuration).

**Reply by fricpto** [reply]
@cromewar  I did everything that you said that I should do and it gave an error 
<img width="742" alt="Screenshot 2022-12-10 171343" src="https://user-images.githubusercontent.com/100000316/206862096-a7b53b7e-60fc-481a-bf3f-205e4df14ea3.png">
and I copied my project ID from Infura
<img width="654" alt="Screenshot 2022-12-10 171501" src="https://user-images.githubusercontent.com/100000316/206862163-1e349323-f489-4467-8c4f-e8f9383fa28f.png">
and it still not working.
This is my code of brownie-config-yaml:
```
dotenv: .env
wallets: 
  from_key: ${PRIVATE_KEY}
```

**Comment by Niferu** [comment]
Hello @fricpto 

You should be using API KEY instead of PROJECT ID, just go to ENDPOINTS tab and you should see your API KEY at the top of Infura website as below:

![image](https://user-images.githubusercontent.com/110537615/207055065-cd16aef5-1371-4b7d-8767-47b57833de80.png)

**Reply by cromewar** [reply]
Does this works @fricpto ?

**Reply by fricpto** [reply]
@cromewar  No, still giving me the same error:
ValueError: Unable to expand environment variable in host setting: 'https://goerli.infura.io/v3/$WEB3_INFURA_PROJECT_ID'

**Reply by cromewar** [reply]
Please try adding the private key to your path trough windows advanced configuration and try again.

**Reply by fricpto** [reply]
@cromewar I have added my private and my API keys to my environment variables and it still the same. But if I wrote "brownie run scripts/deploy.py -network goerli"  in the terminale it will give this:
<img width="515" alt="Screenshot 2022-12-13 232933" src="https://user-images.githubusercontent.com/100000316/207447849-9194b645-ea8b-48a1-a227-97bbbf22e9b5.png">
And if I wrote"brownie run scripts/deploy.py --network goerli" I will get the error "ValueError: Unable to expand environment variable in host setting: 'https://goerli.infura.io/v3/$WEB3_INFURA_PROJECT_ID' "

**Reply by cromewar** [reply]
May be is problem with the naming of your keys, be sure the INFURA key variable name is like this: https://github.com/PatrickAlphaC/brownie_simple_storage/blob/main/.env
And also don't forget to add dotenv to your brownie config as well.

**Reply by Niferu** [reply]
Hi @fricpto 

Please send us your .env file (cover like half of all keys) and brownie-config.yaml I'm pretty sure issue is located in one of them and I think it will be .env incorrectly setup. If you dont want to share partially covered .env just take a look at repo .env example and fix yours accordingly as your error says that there is smth wrong with your INFURA, so that might be INFURA setup incorrectly too.

**Reply by cromewar** [reply]
In addition to what @Neftyr requested also please a screenshoot of your project setup.

**Reply by fricpto** [reply]
@cromewar @Neftyr here are the things that you requested:
<img width="166" alt="Screenshot 2022-12-16 235513" src="https://user-images.githubusercontent.com/100000316/208195689-5c71354a-8356-4c0f-869e-11fb5a557d6e.png">

<img width="638" alt="Screenshot 2022-12-16 235423" src="https://user-images.githubusercontent.com/100000316/208195501-9197a571-857d-473f-9eb3-df109f0b9f91.png">
<img width="207" alt="Screenshot 2022-12-16 235445" src="https://user-images.githubusercontent.com/100000316/208195542-e60882ed-6d41-48a2-aeac-8216612dea0e.png">

**Comment by cromewar** [comment, marked as answer]
Hey @fricpto 
The problem is you configuration file is called `brownie-config-yaml` and it should be `brownie-config.yaml`.
Cheers.

**Reply by fricpto** [reply]
@cromewar  thanks for your help it worked.

**Reply by cromewar** [reply]
I'm glad we were able to solve your problem.
